cli/start.go

```diff
@@ -252,3 +246,97 @@ func MakeStartCommand() *cobra.Command {
 		"Skip generating an encryption key. Encryption at rest will be disabled. WARNING: This cannot be undone.")
 	return cmd
 }
+
+func addEncryptionKey(kr keyring.Keyring, cfg *viper.Viper, opts []node.Option) ([]node.Option, error) {
+	encryptionKey, err := kr.Get(encryptionKeyName)
+	if err != nil && errors.Is(err, keyring.ErrNotFound) && !cfg.GetBool("datastore.noencryption") {
+		encryptionKey, err = crypto.GenerateAES256()
+		if err != nil {
+			return nil, err
+		}
+		err = kr.Set(encryptionKeyName, encryptionKey)
+		if err != nil {
+			return nil, err
+		}
+		log.Info("generated encryption key")
+	} else if err != nil && !errors.Is(err, keyring.ErrNotFound) {
+		return nil, err
+	}
+	opts = append(opts, node.WithBadgerEncryptionKey(encryptionKey))
+	return opts, nil
+}
+
+func addPeerKey(kr keyring.Keyring, opts []node.Option) ([]node.Option, error) {
+	peerKey, err := kr.Get(peerKeyName)
+	if err != nil && errors.Is(err, keyring.ErrNotFound) {
+		peerKey, err = crypto.GenerateEd25519()
+		if err != nil {
+			return nil, err
+		}
+		err = kr.Set(peerKeyName, peerKey)
+		if err != nil {
+			return nil, err
+		}
+		log.Info("generated peer key")
+	} else if err != nil {
+		return nil, err
+	}
+	return append(opts, net.WithPrivateKey(peerKey)), nil
+}
+
+func addIdentity(kr keyring.Keyring, opts []node.Option) ([]node.Option, error) {
+	identityBytes, err := kr.Get(identityKeyName)
+	info := "Loaded identity"
+	if err != nil {
+		if !errors.Is(err, keyring.ErrNotFound) {
+			return nil, err
+		}
+		privateKey, err := crypto.GenerateSecp256k1()
+		if err != nil {
+			return nil, err
+		}
+		identityBytes := privateKey.Serialize()
+		err = kr.Set(identityKeyName, identityBytes)
+		if err != nil {
+			return nil, err
+		}
+		info = "Generated identity"
+	}
+
+	nodeIdentity, err := identity.FromPrivateKey(
+		secp256k1.PrivKeyFromBytes(identityBytes),
+		time.Duration(0),
+		immutable.None[string](),
+		immutable.None[string](),
+		false,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info(info, corelog.Any("Identity", nodeIdentity.IntoRawIdentity()))
+
+	return append(opts, db.WithNodeIdentity(nodeIdentity)), nil
+}
+
+func addEphemeralIdentity(opts []node.Option) ([]node.Option, error) {
+	privateKey, err := crypto.GenerateSecp256k1()
+	if err != nil {
+		return nil, err
+	}
+
+	nodeIdentity, err := identity.FromPrivateKey(
+		secp256k1.PrivKeyFromBytes(privateKey.Serialize()),
+		time.Duration(0),
+		immutable.None[string](),
+		immutable.None[string](),
+		false,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info("Generated ephemeral identity", corelog.Any("Identity", nodeIdentity.IntoRawIdentity()))
```

Comment:
Same as above:
question: Are you sure we want to dump/log the full identity information of the node? i.e. including the private key.

Reply:
removed